src/nvim/CMakeLists.txt

```diff
@@ -189,8 +192,14 @@ list(APPEND NVIM_LINK_LIBRARIES
     ${LIBUV_LIBRARIES}
     ${MSGPACK_LIBRARIES}
     ${LUAJIT_LIBRARIES}
+    "-Wl,--whole-archive"
```

Comment:
I should add that the error above is on Yosemite where it is using LLVM's linker.  Mavericks is probably a different story, but FreeBSD may fail as well.  There is a `-all_load`, but the corresponding `-noall_load` is obsolete.  Turns out `-all_load` affects all static archives. :-(  There is also a `-force_load LIBRARY`, but detecting the presence of the option at build time is kind of a pain since it requires an archive.